hawc/apps/animal/serializers.py

```diff
@@ -295,14 +297,62 @@ def to_representation(self, instance):
 
         return ret
 
+    def _validate_term_and_text(self, data, term_field: str, text_field: str, term_type_str: str):
```

Comment:
Rewrote the field-level validation and made it instance-level since the validation is more than single-dependent. Also made it easier to cache and update things